Src/Plugins/Security/SEC101_011.SquareCredentialsValidator.cs

```diff
@@ -43,5 +60,74 @@ protected override IEnumerable<ValidationResult> IsValidStaticHelper(Dictionary<
 
             return new[] { validationResult };
         }
+
+        protected override ValidationState IsValidDynamicHelper(ref Fingerprint fingerprint,
+                                                                ref string message,
+                                                                Dictionary<string, string> options,
+                                                                ref ResultLevelKind resultLevelKind)
+        {
+            string id = fingerprint.Id;
```

Comment:
>fingerprint [](http://example.com/codeflow?start=24&length=11)

It looks like this code doesn't return any asset information, i.e., the identifier for the compromised thing. Please see other rules for examples of proper reporting. I think 'id' is the thing we should be surfacing that we are not.

Please make sure to review actual user-facing strings returned by this change.

@eddynaka, enough is enough: we need proper unit and functional testing that forces testing through the dynamic validation piece. :)

Reply:
We need to report the id for this code path but can delay getting the additional testing out there. But the very next thing we should do is implement an approach for this.

Basically, for every dynamic validator, we need a way to exercise/validate all possible code paths. This will be tricky/likely require a mocked HTTP client/other things to accomplish.

Reply:
Added asset information to retrun messages.

Reply:
Can you provide an example of what the user-facing string looks like for authorized/unauthorized access? I think there's another overload for some helpers that allow you to provide a descriptor of the id, when helpful. to clarify the output. So looking at current example would help understand whether we need to do something like that.

Reply:
@michaelcfanning 

For successfully validating the response is 
> The compromised asset is 'a'

For a credential that doesn't validate it's
> The provided secret is not authorized to access 'a'.

In both cases 'a' is replaced with the `client_id` of the Square app that was detected. (which normally looks like `sq0idp-<alphanumeric-identifier>`